src/Adapter/MSTest.CoreAdapter/Execution/TypeCache.cs

```diff
@@ -635,9 +636,47 @@ private TestMethodAttribute GetTestMethodAttribute(MethodInfo methodInfo, TestCl
         /// <returns> The <see cref="MethodInfo"/>. </returns>
         private MethodInfo GetMethodInfoForTestMethod(TestMethod testMethod, TestClassInfo testClassInfo)
         {
-            var methodsInClass = testClassInfo.ClassType.GetRuntimeMethods().ToArray();
             MethodInfo testMethodInfo;
 
+            if (testMethod.HasManagedMethodAndType)
+            {
+                var assembly = testClassInfo.ClassType.GetTypeInfo().Assembly;
```

Comment:
To help with perf, could we pass in the assembly information to this method?

Reply:
or cache assembly info.

Reply:
We probably don't need `testClassInfo` at all. I'll check and refactor this.